### PR TITLE
EXL-318 - Created tool for scanning imported files

### PR DESCRIPTION
**Pull request by zsso**
- Create custom CodeSniffer sniff to scan and replace old class names

### CODE REVIEW

**Review by hupupu — CHANGES_REQUESTED** [review]

File: plugin/front/scanImportedFile.php [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        'PluginFieldsPrintercustomfield' => 'PluginFieldsPrinterprintercustomfield',
+        'PluginFieldsContractcustomfield' => 'PluginFieldsContractcontractcustomfield',
+        'PluginFieldsCartridgeitemcustomfield' => 'PluginFieldsCartridgeitemcartridgeitemcustomfield',
+        'PluginFieldsCartridgecustomfield' => 'PluginFieldsCartridgecartridgecustomfield',
+        'clickable' => 'pointer',
+        'getFromDBByItemsId' => 'populateByItemsId???',
+        'getFromDBByQuery' => 'populateByQuery???',
+        'include (\'../../../inc/includes.php\')' => 'require \'../inc/includes.php\';',
+        '$CFG_GLPI[root_doc]/plugins/iservice' => '$CFG_PLUGIN_ISERVICE[root_doc]',
+        'DIRECTORY_SEPARATOR' => "'/'",
```
[the remark this answers is not in the document]
Comment (zsso) [reply plugin/front/scanImportedFile.php:22]: Needs manual handling... Please see comment above.

File: plugin/front/scanImportedFile.php [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    ];
+
+    $customFieldDefinitionFiles = glob(PLUGIN_ISERVICE_DIR . '/install/customfields/*.json');
+
+    foreach ($customFieldDefinitionFiles as $customFieldDefinitionFile) {
+        $fieldMap = json_decode(file_get_contents($customFieldDefinitionFile), true);
+        foreach ($fieldMap as $field) {
+            if (!isset($field['type']) || $field['type'] == 'dropdown' || $field['type'] == 'header') {
+                continue;
+            }
```
[the remark this answers is not in the document]
Comment (zsso) [reply plugin/front/scanImportedFile.php:32]: Mistake.

File: plugin/front/scanImportedFile.php [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        'PluginIserviceCommon::getQueryResult' => 'PluginIserviceDB::getQueryResult',
+        'PluginIserviceCommon' => 'IserviceToolBox',
+        'PluginFieldsSuppliercustomfield' => 'PluginFieldsSuppliersuppliercustomfield',
+        'PluginFieldsPrintermodelcustomfield' => 'PluginFieldsPrintermodelprintermodelcustomfield',
+        'PluginFieldsTicketcustomfield' => 'PluginFieldsTicketticketcustomfield',
+        'PluginFieldsPrintercustomfield' => 'PluginFieldsPrinterprintercustomfield',
+        'PluginFieldsContractcustomfield' => 'PluginFieldsContractcontractcustomfield',
+        'PluginFieldsCartridgeitemcustomfield' => 'PluginFieldsCartridgeitemcartridgeitemcustomfield',
+        'PluginFieldsCartridgecustomfield' => 'PluginFieldsCartridgecartridgecustomfield',
+        'clickable' => 'pointer',
```
[the remark this answers is not in the document]
Comment (hupupu) [reply plugin/front/scanImportedFile.php:17]: Exactly this is why I added the `???`

**Review by hupupu — APPROVED** [review]

File: plugin/front/scanImportedFile.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (zsso) [reply plugin/front/scanImportedFile.php:17]: I see, I was confused. As I added '???' to some other values, I thought that I added it and you are asking what the '???' means...

**Pull request merged**